File: leetcode_657/solution.go
package main

import "fmt"

func judgeCircle(moves string) bool {
	x, y := 0, 0
	for _, step := range moves {
		switch string(step) {
		case "L":
			x -= 1
		case "R":
			x += 1
		case "U":
			y += 1
		case "D":
			y -= 1
		}
	}
	return (x == 0) && (y == 0)
}

func main() {
	moves := "UDDD"
	ret := judgeCircle(moves)
	fmt.Println(ret)
}

File: leetcode_657/solution.js
var judgeCircle = function(moves) {
    let x = 0, y = 0;
    [...moves].map((step) => {
        switch (step) {
            case "L": x -= 1; break;
            case "R": x += 1; break;
            case "U": y += 1; break;
            case "D": y -= 1; break;
        }
    });
    return (x === 0) && (y === 0);
};

function main () {
    moves = "UDDD";
    ret = judgeCircle(moves);
    console.log(ret);
}

main()